src/extensions/cellpack/model.ts

```diff
@@ -477,30 +506,57 @@ async function loadMembrane(plugin: PluginContext, name: string, state: State, p
-        .apply(StateTransforms.Model.ModelFromTrajectory, undefined, { state: { isGhost: true } })
-        .apply(StructureFromAssemblies, undefined, { state: { isGhost: true } })
-        .commit({ revertOnError: true });
-
-    const membraneParams = {
-        representation: params.preset.representation,
+    const props = {
+        type: {
+            name: 'assembly' as const,
+            params: { id: '1' }
+        }
     };
-
-    await CellpackMembranePreset.apply(membrane, membraneParams, plugin);
+    if (params.source.name === 'id' && params.source.params !== 'MycoplasmaGenitalium.json'){
+        // old membrane
+        const membrane = await b.apply(StateTransforms.Data.ParseCif, undefined, { state: { isGhost: true } })
+            .apply(StateTransforms.Model.TrajectoryFromMmCif, undefined, { state: { isGhost: true } })
+            .apply(StateTransforms.Model.ModelFromTrajectory, undefined, { state: { isGhost: true } })
+            .apply(StructureFromAssemblies, undefined, { state: { isGhost: true } })
+            .commit({ revertOnError: true });
+        const membraneParams = {
+            representation: params.preset.representation,
+        };
+        await CellpackMembranePreset.apply(membrane, membraneParams, plugin);
+    } else {
+        const membrane = await b.apply(StateTransforms.Data.ParseCif, undefined, { state: { isGhost: true } })
+            .apply(StateTransforms.Model.TrajectoryFromMmCif, undefined, { state: { isGhost: true } })
+            .apply(StateTransforms.Model.ModelFromTrajectory, undefined, { state: { isGhost: true } })
+            .apply(StateTransforms.Model.StructureFromModel, props, { state: { isGhost: true } })
+            .commit({ revertOnError: true });
+        const membraneParams = {
+            representation: params.preset.representation,
+        };
+        await CellpackMembranePreset.apply(membrane, membraneParams, plugin);
+    }
 }
 
 async function loadPackings(plugin: PluginContext, runtime: RuntimeContext, state: State, params: LoadCellPackModelParams) {
     const ingredientFiles = params.ingredients || [];
 
     let cellPackJson: StateBuilder.To<PSO.Format.Json, StateTransformer<PSO.Data.String, PSO.Format.Json>>;
+    let modelFile: Asset.File|null = params.results;
     if (params.source.name === 'id') {
         const url = Asset.getUrlAsset(plugin.managers.asset, getCellPackModelUrl(params.source.params, params.baseUrl));
+        // console.log("getting "+params.source.params+" "+url.url);
         cellPackJson = state.build().toRoot()
             .apply(StateTransforms.Data.Download, { url, isBinary: false, label: params.source.params }, { state: { isGhost: true } });
+
+        if (params.source.params === 'MycoplasmaGenitalium.json'){
+            const m_url = Asset.getUrlAsset(plugin.managers.asset, `${params.baseUrl}/results/results_149_curated_serialized.bin`);
```

Comment:
How about putting the URL to the .bin file into the .json file and resolve it in `StructureFromCellpack` instead of passing it as a parameter?

The .bin file should retrieved using the `plugin.managers.asset`
`const asset = await plugin.runTask(plugin.managers.asset.resolve(Asset.getUrlAsset(assetManager, url), 'binary'));`. The resulting `asset` needs to be added to the `assets` variable in `StructureFromCellpack.apply`. That way the bin file will be part of the .molx session file.

Reply:
I moved the code in the model.ts file. However I am not sure how to add the asset to the asset manager 
`        if (options.resultfile) {
            const asset = await plugin.runTask(plugin.managers.asset.resolve(Asset.getUrlAsset(plugin.managers.asset, `${params.baseUrl}/results/${options.resultfile}`), 'binary', true));
            modelFile = Asset.File(new File([asset.data], 'model.bin'));
        }`

Reply:
Getting the asset using `plugin.managers.asset.resolve` already adds it. That part is good and it will be part of the .molx session. What is missing is the ensuring that the asset gets disposed of when the transform is deleted from the state tree, see https://github.com/corredD/molstar-proto/blob/393fc99ed2f717f844bb738b269a63f2910c6870/src/extensions/cellpack/state.ts#L132-L140.

They way you are doing the bin packing now does not yet work with sessions. Currently you are doing all the bin packing loading as part of an the LoadCellPackModel Action which means the bin packing data is not added to the state and thus not saved in the session. You can try to add the bin packing parsing as part of the ParseCellPack Transform.